### PR TITLE
chore(release): pull release/v1.50.0 into main

### PR DESCRIPTION
**Pull request by devops-github-rudderstack**
:crown: *An automated PR*

### CODE REVIEW

**Review by sanpj2292 — APPROVED** [review]

File: src/v0/destinations/facebook_conversions/utils.js [hunk 1]
```diff
@@ -44,7 +44,11 @@ const getCategoryFromEvent = (eventName) => {
...
 
-const populateContentsAndContentIDs = (productPropertiesArray, fallbackQuantity) => {
+const populateContentsAndContentIDs = (
+  productPropertiesArray,
+  fallbackQuantity,
+  fallbackDeliveryCategory,
+) => {
   const contentIds = [];
   const contents = [];
   if (Array.isArray(productPropertiesArray)) {
```
Comment (krishna2020) [inline src/v0/destinations/facebook_conversions/utils.js:54]: @sandeepdsvs  with check of `isObject(productProps)`, Does contentIds & contents gets loaded for `product added to wishlist` ?
https://github.com/rudderlabs/rudder-transformer/blob/2055dabb25e0e67a61dda263ca75f93408f1bf9c/src/v0/destinations/facebook_conversions/utils.js#L129

File: src/v0/destinations/salesforce/utils.js
Comment (krishna2020) [inline src/v0/destinations/salesforce/utils.js:1]: @shrouti1507 unit tests for util class is missing

File: src/v0/destinations/facebook_conversions/utils.js (hunk 1 above)
Comment (sandeepdsvs) [reply src/v0/destinations/facebook_conversions/utils.js:54]: Yes KC, they do get loaded.

File: src/v0/destinations/salesforce/utils.js [hunk 2]
```diff
@@ -140,4 +156,30 @@ const getAccessToken = async (destination) => {
   });
 };
 
-module.exports = { getAccessToken, salesforceResponseHandler };
+const collectAuthorizationInfo = async (event) => {
+  let authorizationFlow;
+  let authorizationData;
+  if (isDefinedAndNotNull(event.metadata?.secret)) {
```
Comment (krishna2020) [inline src/v0/destinations/salesforce/utils.js:162]: Is there no corner scenario where `event.metadata?.secret` might come as null for oauth destination ?
Instead why not make use of destDef to have clear segregation ? Or have salesforce_oauth/transform.js just as wrapper class setting `authorizationFlow=OAUTH` & re-use the existing code ?

File: src/v0/destinations/salesforce/utils.js [hunk 3]
```diff
@@ -140,4 +156,30 @@ const getAccessToken = async (destination) => {
...
+  const { authorizationFlow, authorizationData } = authInfo;
+  return authorizationFlow === OAUTH
+    ? { Authorization: `Bearer ${authorizationData.token}` }
+    : { Authorization: authorizationData.token };
+};
+
+module.exports = {
+  getAccessTokenOauth,
+  salesforceResponseHandler,
+  getAccessToken,
```
Comment (krishna2020) [inline src/v0/destinations/salesforce/utils.js:182]: can remove export of `getAccessToken` if not used in transform.js

File: src/v0/destinations/facebook_conversions/utils.js (hunk 1 above)
Comment (sandeepdsvs) [reply src/v0/destinations/facebook_conversions/utils.js:54]: for `product added to wishlist` `productPropertiesArray` will be of length 1 and `message.properties`, if present, will pass `isObject(productProps)` condition

**Review by krishna2020 — APPROVED** [review]

**Review by aashishmalik — APPROVED** [review]

**Review by Jayachand — APPROVED** [review]

File: src/v0/destinations/salesforce/utils.js
Comment (shrouti1507) [reply src/v0/destinations/salesforce/utils.js:1]: added two very very small util functions. The entire pre-existing file does not have unit test cases. Will add it when salesforce v2 is released.

**Pull request merged**